sdk/opa_test.go

```diff
@@ -357,6 +359,172 @@ main {
 	}
 }
 
+func TestDecisionWithMetrics(t *testing.T) {
+
+	ctx := context.Background()
+
+	server := sdktest.MustNewServer(
+		sdktest.MockBundle("/bundles/bundle.tar.gz", map[string]string{
+			"main.rego": `
+package system
+
+main = true
+`,
+		}),
+	)
+
+	defer server.Stop()
+
+	config := fmt.Sprintf(`{
+		"services": {
+			"test": {
+				"url": %q
+			}
+		},
+		"bundles": {
+			"test": {
+				"resource": "/bundles/bundle.tar.gz"
+			}
+		}
+	}`, server.URL())
+
+	opa, err := sdk.New(ctx, sdk.Options{
+		Config: strings.NewReader(config),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer opa.Stop(ctx)
+
+	m := metrics.New()
+	if result, err := opa.Decision(ctx, sdk.DecisionOptions{
+		Metrics: m,
+	}); err != nil {
+		t.Fatal(err)
+	} else if decision, ok := result.Result.(bool); !ok || !decision {
+		t.Fatal("expected true but got:", decision, ok)
+	}
+
+	if exp, act := 4, len(m.All()); exp != act {
+		t.Fatalf("expected %d metrics, got %d", exp, act)
+	}
+
+	expectedRecordedMetrics := map[string]bool{
+		"timer_rego_query_compile_ns": true,
+		"timer_rego_query_eval_ns":    true,
+		"timer_rego_query_parse_ns":   true,
+		"timer_sdk_decision_eval_ns":  true,
+	}
+	for k := range m.All() {
+		if ok := expectedRecordedMetrics[k]; !ok {
+			t.Errorf("unexpected metric %s recorded", k)
+		}
+	}
+
```

Comment:
Nicely done. If it's got 4, and nothing unexpected, it must have the correct ones. ✔️